packages/lab/src/dialog/Dialog.tsx

```diff
@@ -119,39 +135,39 @@ export const Dialog = forwardRef<HTMLDivElement, DialogProps>(function Dialog(
 
   return (
     <DialogContext.Provider value={contextValue}>
-      {showComponent && (
-        <Scrim fixed>
-          <FloatingComponent
-            open={open}
-            role={role}
-            aria-modal="true"
-            ref={floatingRef}
-            focusManagerProps={{
-              context: context,
-            }}
-            aria-labelledby={`${id}-header`}
-            className={clsx(
-              withBaseName(),
-              withBaseName(size, currentbreakpoint),
-              {
-                [withBaseName("enterAnimation")]: open,
-                [withBaseName("exitAnimation")]: !open,
-                [withBaseName(status as string)]: status,
-              },
-              className
-            )}
-            onAnimationEnd={() => {
-              if (!open && showComponent) {
-                setShowComponent(false);
-              }
-            }}
-            {...getFloatingProps()}
-            {...rest}
-          >
-            {children}
-          </FloatingComponent>
-        </Scrim>
-      )}
+      <ConditionalScrimWrapper condition={open && !disableScrim}>
+        <FloatingComponent
+          open={showComponent}
+          role={role}
```

Comment:
role usage here can be simplified by hardcoding it here: role="dialog" and allowing role to be spread below by removing it from being destructured. Also we need to look at useRole again. I didn't realise floating-ui will change things depending on what value you provide it so need to double check it doesn't do anything special with dialog.

Reply:
when role='alert dialog' is passed will both 'dialog' and 'alert dialog' roles then not be attached to the same element ?
This no longer uses the useRole hook from floating-ui, you thinking we should be added back in ?